### ISSUE TITLE
RuntimeError: /pytorch/torch/csrc/jit/fuser/cuda/fused_kernel.cpp:202: misaligned address

### ISSUE THREAD

**Issue by msieb1** [pytorch/pytorch#31935]
## 🐛 Bug
Using a model that was traced via `torch.jit.trace`, the traced forward call of the model produces the following error:

`RuntimeError: /pytorch/torch/csrc/jit/fuser/cuda/fused_kernel.cpp:202: misaligned address`

Update:
I noticed this error only happens when using Float16 instead of Float32.

cc @suo

**Comment by eellison** [comment]
fuser error, cc @mruberry @jamesr66a

**Comment by ngimel** [comment]
Can you please provide a script that would reproduce this error?

**Comment by msieb1** [comment]
I was able to track the error down to torch.multinomial not support torch.float16 for PyTorch 1.1.0. Running float16 through torch.multinomial via torch.jit.trace produced the misaligned error.

**Comment by mruberry** [comment]
Can you produce a small, self-contained python snippet that we can run to observe the same error you're seeing?

**Comment by miquel-espinosa** [comment]
I am also encountering the error: `RuntimeError: CUDA error: misaligned address`.
It happens when float16 is used together with multiple GPUs. 

With more detail:

- To enable training on multiple GPUs, I am using nn.DataParallel().
- To use half precision, I am using torch.cuda.amp.GradScaler().

When both (multi-GPU and float16) used simultaneously, it yields the mentioned error.
(If used independently, no error is thrown).

**Comment by mruberry** [comment]
> I am also encountering the error: `RuntimeError: CUDA error: misaligned address`. It happens when float16 is used together with multiple GPUs.
> 
> With more detail:
> 
> * To enable training on multiple GPUs, I am using nn.DataParallel().
> * To use half precision, I am using torch.cuda.amp.GradScaler().
> 
> When both (multi-GPU and float16) used simultaneously, it yields the mentioned error. (If used independently, no error is thrown).

I'm sorry to hear you're hitting that issue, @miquel-espinosa -- would you file a new issue describing your problem and include some example code we can use to reproduce it? It sounds like a different problem

**Comment by miquel-espinosa** [comment]
Sorry for the late reply.

I tried to reproduce the error with a simple vanilla network, but the error did not arise in such case.
So I suspected that it must have been thrown from a code error or some hardware specific thing.

But later on, I decreased the batch size and the error does not appear now, and I'm able to train correctly the model.
Therefore, I think it was probably an issue with the memory, that it run out of memory or something like that.

Thanks for the support!
